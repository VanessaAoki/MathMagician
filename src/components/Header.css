header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.8em 4em;
  background-color: var(--orange);
  color: white;
}

header ul {
  list-style: none;
  display: flex;
}

header ul li {
  padding: 0 0.5em;
}

header ul li:not(:last-child) {
  border-right: 1px solid white;
}

.highlight {
  font-weight: bold;
  text-decoration: underline;
}
