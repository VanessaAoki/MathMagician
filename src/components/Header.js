import React from 'react';
import {
  Link, NavLink,
} from 'react-router-dom';
import './Header.css';

const Header = () => (
  <header>
    <h1><Link to="/home">Math Magicians</Link></h1>
    <nav>
      <ul id="navbar">
        <li><NavLink activeClassName="highlight" to="/home">Home</NavLink></li>
        <li><NavLink activeClassName="highlight" to="/calculator">Calculator</NavLink></li>
        <li><NavLink activeClassName="highlight" to="/quote">Quote</NavLink></li>
      </ul>
    </nav>
  </header>
);

export default Header;
