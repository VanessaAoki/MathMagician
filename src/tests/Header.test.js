import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header';

const tree = renderer.create(
  <MemoryRouter>
    <Header />
  </MemoryRouter>,
).toJSON();

it('Header UI renders as expected, we have the logo, quote, calculator and home links', () => {
  expect(tree).toMatchSnapshot();
});
