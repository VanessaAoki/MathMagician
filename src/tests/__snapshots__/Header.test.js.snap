// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header UI renders as expected, we have the logo, quote, calculator and home links 1`] = `
<header>
  <h1>
    <a
      href="/home"
      onClick={[Function]}
    >
      Math Magicians
    </a>
  </h1>
  <nav>
    <ul
      id="navbar"
    >
      <li>
        <a
          aria-current={null}
          href="/home"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
